lib/common/database.dart

```diff
@@ -16,16 +16,13 @@ class DatabaseService {
   FirebaseFirestore.instance.collection('users');
 
   factory DatabaseService({String? uid}) {
-    if (uid != null) {
-      userid = uid;
-    }
     return _singleton;
   }
 
   DatabaseService._internal();
```

Comment:
What does this do? I see _singleton is assigned it but not sure what it is

Reply:
Basically [how to do a singleton factory constructor](https://stackoverflow.com/questions/12649573/how-do-you-build-a-singleton-in-dart). From my understanding it relies on the static variable itself by constructing itself just once, and factory to specify to not build itself each call.

Reply:
![image](https://github.com/Scrumbags115/plannertarium/assets/130195350/20355697-236d-420b-a569-01adc05d53fb)

would something like this work? or does that break the singleton thing

Reply:
that's just a normal class though? since `Database.factory()` is supposed to return the singleton, and it's being thrown away since it's not being returned and DatabaseService is the normal constructor.

Reply:
ahh ok sorry I misunderstood what that was.